src/query/service/src/table_functions/cloud/task_history.rs

```diff
@@ -0,0 +1,320 @@
+    pub fn create(
+        database_name: &str,
+        table_func_name: &str,
+        table_id: u64,
+        table_args: TableArgs,
+    ) -> Result<Arc<dyn TableFunction>> {
+        let args_parsed = TableHistoryArgsParsed::parse(&table_args)?;
+        let table_info = TableInfo {
+            ident: TableIdent::new(table_id, 0),
+            desc: format!("'{}'.'{}'", database_name, table_func_name),
+            name: String::from("task_history"),
+            meta: TableMeta {
+                schema: infer_table_schema(&task_run_schema())
+                    .expect("failed to infer table schema"),
+                engine: String::from(table_func_name),
+                // Assuming that created_on is unnecessary for function table,
+                // we could make created_on fixed to pass test_shuffle_action_try_into.
+                created_on: DateTime::from_timestamp(0, 0).unwrap(),
+                updated_on: DateTime::from_timestamp(0, 0).unwrap(),
+                ..Default::default()
+            },
+            ..Default::default()
+        };
+
+        Ok(Arc::new(TaskHistoryTable {
+            table_info,
+            args_parsed,
+            table_args,
+        }))
+    }
+}
+
+#[async_trait::async_trait]
+impl Table for TaskHistoryTable {
+    fn is_local(&self) -> bool {
+        true
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    #[async_backtrace::framed]
+    async fn read_partitions(
+        &self,
+        _ctx: Arc<dyn TableContext>,
+        _push_downs: Option<PushDownInfo>,
+        _dry_run: bool,
+    ) -> Result<(PartStatistics, Partitions)> {
+        // dummy statistics
+        Ok((PartStatistics::new_exact(1, 1, 1, 1), Partitions::default()))
+    }
+
+    fn table_args(&self) -> Option<TableArgs> {
+        Some(self.table_args.clone())
+    }
+
+    fn read_data(
+        &self,
+        ctx: Arc<dyn TableContext>,
+        _plan: &DataSourcePlan,
+        pipeline: &mut Pipeline,
+        _put_cache: bool,
+    ) -> Result<()> {
+        pipeline.add_source(
+            |output| TaskHistorySource::create(ctx.clone(), output, self.args_parsed.clone()),
+            1,
+        )?;
+
+        Ok(())
+    }
+}
+
+struct TaskHistorySource {
+    is_finished: bool,
+    args_parsed: TableHistoryArgsParsed,
+    ctx: Arc<dyn TableContext>,
+}
+
+impl TaskHistorySource {
+    pub fn create(
+        ctx: Arc<dyn TableContext>,
+        output: Arc<OutputPort>,
+        args_parsed: TableHistoryArgsParsed,
+    ) -> Result<ProcessorPtr> {
+        AsyncSourcer::create(ctx.clone(), output, TaskHistorySource {
+            ctx,
+            args_parsed,
+            is_finished: false,
+        })
+    }
+    async fn build_request(&self) -> Result<ShowTaskRunsRequest> {
+        let tenant = self.ctx.get_tenant();
+        let available_roles = self.ctx.get_available_roles().await?;
```

Comment:
TODO: https://github.com/datafuselabs/databend/pull/15458/files task privilege need consider this new table function